src/core/tensor/tensor.cc

```diff
@@ -161,6 +161,9 @@ Tensor Resize(const Tensor &in, const Shape &shape) {
 
 // return new tensor
 Tensor Tensor::AsType(const DataType type) {
+  CHECK(block() && block()->initialized() == true)
+      << "the data of the tensor needs be initialized before casting to "
```

Comment:
this cannot fix the error. need to update the test case to reverse the two lines.

Reply:
Done.

Reply:
There seem to be many such cases in unit testing